serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -49,147 +53,171 @@
+     * @param properties        - parameters needed to contact the provided services
+     * @param virksertService   - used to lookup virksomhetssertifikat (certificate)
      * @param elmaLookupService - used to lookup hostname of Altinn formidlingstjeneste
-     * @param krrService - used to lookup parameters needed to use DPI transportation
+     * @param krrService        - used to lookup parameters needed to use DPI transportation
+     * @param entityService     - used to look up information about citizens and organizations in Brønnøysundregisteret and Datahotellet.
+     * @param svarUtService     - used to determine whether an organization utilizes FIKS.
+     * @param documentTypeService
+     * @param requestScope
      */
-    @Autowired
     public ServiceRecordFactory(ServiceregistryProperties properties,
                                 VirkSertService virksertService,
                                 ELMALookupService elmaLookupService,
                                 KrrService krrService,
                                 EntityService entityService,
-                                SvarUtService svarUtService) {
+                                SvarUtService svarUtService,
+                                ProcessService processService,
+                                DocumentTypeService documentTypeService,
+                                SRRequestScope requestScope) {
         this.properties = properties;
         this.virksertService = virksertService;
         this.elmaLookupService = elmaLookupService;
         this.krrService = krrService;
         this.entityService = entityService;
         this.svarUtService = svarUtService;
+        this.processService = processService;
+        this.documentTypeService = documentTypeService;
+        this.requestScope = requestScope;
     }
 
-    @HystrixCommand(fallbackMethod = "createFiksErrorRecord")
-    public Optional<FiksWrapper> createFiksServiceRecord(String orgnr) {
-        Optional<Integer> adresseringNivaa = svarUtService.hasSvarUtAdressering(orgnr);
-        if (adresseringNivaa.isPresent()) {
-            String pem;
-            try {
-                pem = IOUtils.toString(properties.getSvarut().getCertificate().getInputStream(), StandardCharsets.UTF_8);
-            } catch (IOException e) {
-                log.error("Could not read certificate from {}", properties.getSvarut().getCertificate().toString());
-                throw new ServiceRegistryException(e);
-            }
-            FiksServiceRecord fiksServiceRecord = new FiksServiceRecord(orgnr, pem, properties.getSvarut().getServiceRecordUrl().toString());
-            return Optional.of(FiksWrapper.of(fiksServiceRecord, adresseringNivaa.get()));
+    public Optional<ServiceRecord> createArkivmeldingServiceRecord(String orgnr, String processIdentifier, Integer targetSecurityLevel) throws SecurityLevelNotFoundException, CertificateNotFoundException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: no.difi.meldingsutveksling.serviceregistry.exceptions.SecurityLevelNotFoundException, no.difi.meldingsutveksling.serviceregistry.CertificateNotFoundException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)